runtime/drivers/duckdb/duckdb.go

```diff
@@ -103,31 +121,70 @@ type connection struct {
 	dbErr       error
 }
 
+// Driver implements drivers.Connection.
+func (c *connection) Driver() string {
+	if val, ok := c.driverConfig["driver"]; ok {
+		return val.(string)
+	}
+	return "duckdb"
+}
```

Comment:
2. I felt even driver name is also a config. May be it will be more explicit once we move to parsing it via `mapstructure`. But we could pass name as parameter as well. What do you think ?

Reply:
Since the caller passes it as a separate parameter, not as part of the config, it feels like it should also be a separate parameter elsewhere (otherwise, it's easy to forget passing it in the config also).

An alternative would be to have separate drivers passed to `drivers.Register`, like `drivers.Register("motherduck", motherduckDriver)`, which then shares the same implementation, but set different driver names.

Probably, just passing the driver name to `Open` is easiest

Reply:
I liked the alternative better since name in all cases would be the same. `name` can just be a field in a driver which can then be passed to connection as driver name.